trickle/trickle_server.go

```diff
@@ -408,9 +413,13 @@ func (s *Stream) getForWrite(idx int) (*Segment, bool) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 	if idx == -1 {
-		idx = s.latestWrite
-	} else {
-		s.latestWrite = idx
+		if s.latestWrite <= 0 {
+			// no writes yet
+			idx = 0
```

Comment:
What if write for the segment 0 has already started, then a post comes with `-1` index? won't this cause 2 writes to the segment 0 instead of 0 and 1? Maybe `latestWrite` could be initialized as `-1` and here we check `< 0` instead 🤔